source/docs/software/wpilib-tools/path-planning/wpilib-trajectories/trajectorygeneration.rst

```diff
@@ -0,0 +1,113 @@
+
+.. tabs::
+
+   .. tab:: Java
+
+      .. literalinclude:: examples/trajectorygeneration-1/java/ExampleSpline.java
+         :language: java
+         :lines: 8-16
+
+   .. tab:: C++
+
+      .. literalinclude:: examples/trajectorygeneration-1/cpp/ExampleSpline.cpp
+         :language: cpp
+         :lines: 7-16
+
+``SplineHelper`` returns a list of splines, with each element representing the spline between one waypoint and another. Splines are used as a tool to generate trajectories; however, the spline itself does not have any information about velocities and accelerations. In order to retrieve this info, a *trajectory* must be generated.
+
+Trajectories
+------------
+You do not have to go through the steps above if you want a trajectory with velocities and accelerations -- the steps above are simply used to generate the curve. The method outlined below will perform the steps mentioned above automatically. The ``TrajectoryGenerator`` class is used to generate trajectories.
+
+The method used to generate a trajectory is ``generateTrajectory(...)``. There are two overloads for this method. One that uses clamped cubic splines and the other uses quintic splines. The overload that uses clamped cubic splines accepts a start waypoint, a list of interior waypoints, and an end waypoint whereas the overload that uses quintic splines accepts a list of ``Pose2d``, representing all waypoints in the trajectory. Both of these overloads also take in the following:
+
+* ``constraints``: This is a vector of user defined constraints to limit the velocity and acceleration along the trajectory.
+* ``startVelocity``: This is the velocity at which the trajectory will start. This should be zero if you want the robot to start from rest.
+* ``endVelocity``: This is the velocity at which the trajectory will end. This should be zero if you want the robot to stop completely at the end of the trajectory.
+* ``maxVelocity``: This is the maximum velocity that the robot should reach when traversing the trajectory.
+* ``maxAcceleration``: This is the maximum acceleration that the robot should experience when traversing the trajectory.
+* ``reversed``: This is whether the robot should travel backwards or not.
+
+Note: The ``reversed`` property simply represents whether the robot is traveling backward. If you specify four waypoints, a, b, c, and d, the robot will still travel in the same order through the waypoints when the ``reversed`` flag is set to ``true``. This also means that you must account for the direction of the robot when providing the waypoints. For example, if your robot is facing your alliance station wall and travels backwards to some field element, the starting waypoint should have a rotation of 180 degrees.
+
+Here is an example of generating a trajectory using clamped cubic splines for the 2018 game, FIRST Power Up:
+
+.. tabs::
+
+   .. tab:: Java
+
+      .. literalinclude:: examples/trajectorygeneration-2/java/ExampleTrajectory.java
```

Comment:
In this example, what are the numbers inside of the Units.feetToMeters are for? What are these constants?

Reply:
These are waypoints for generating the trajectory. They are just locations on the field with the bottom right corner of the field being the origin.